src/remote_adapter/remote_adapter_server/plugins/utils/utils_manager.h

```diff
@@ -0,0 +1,152 @@
+#include <signal.h>
+#include <string>
+#include <sys/procfs.h>
+#include <vector>
+
+namespace utils_wrappers {
+
+#ifndef __QNX__
+typedef int procfs_info;
+#endif
+
+class UtilsManager : public remote_adapter::UtilsPlugin {
+public:
+  void Bind(rpc::server &server) override;
+  std::string PluginName() override;
+
+  /*
+   * @brief Run of application
+   *
+   * @param app_path full path to the application folder
+   * @param app_name app name
+   * @return code from error_codes namespace, SUCCESS or FAILED
+   */
+  static int StartApp(const std::string &app_path, const std::string &app_name);
+  /*
+   * @brief Stop of application
+   *
+   * @param app_name app name
+   * @return code from error_codes namespace, SUCCESS or FAILED
+   */
+  static int StopApp(const std::string &app_name, const int sig = SIGTERM);
+  /*
+   * @brief Check application status
+   *
+   * @param app_name app name
+   * @return code from stat_app_codes namespace, NOT_RUNNING,RUNNING or CRASHED
+   */
+  static int CheckAppStatus(const std::string &app_name);
+  /*
+   * @brief Create backup of given file
+   *
+   * @param file_path full path to the file folder
+   * @param file_name file name
+   * @return code from error_codes namespace, SUCCESS or FAILED
+   */
+  static int FileBackup(const std::string &file_path,
+                        const std::string &file_name);
+  /*
+   * @brief Restore backup of given file
+   *
+   * @param file_path full path to the file folder
+   * @param file_name file name
+   * @return code from error_codes namespace, SUCCESS or FAILED
+   */
+  static int FileRestore(const std::string &file_path,
+                         const std::string &file_name);
+  /*
+   * @brief Update file contents
+   *
+   * @param file_path full path to the file folder
+   * @param file_name file name
+   * @param file_content file contents
+   * @return code from error_codes namespace, SUCCESS or FAILED
+   */
+  static int FileUpdate(const std::string &file_path,
+                        const std::string &file_name,
+                        const std::string &file_content);
+  /*
+   * @brief Check file availability
+   *
+   * @param file_path full path to the file folder
+   * @param file_name file name
+   * @return code from error_codes namespace,  SUCCESS - if file exist,
+   * otherwise FAILED
+   */
+  static int FileExists(const std::string &file_path,
+                        const std::string &file_name);
+  /*
+   * @brief File deletion
+   *
+   * @param file_path full path to the file folder
+   * @param file_name file name
+   * @return code from error_codes namespace, SUCCESS if successful, otherwise
+   * FAILED
+   */
+  static int FileDelete(const std::string &file_path,
+                        const std::string &file_name);
+  /*
+   * @brief Get file contents
+   *
+   * @param file_path full path to the file folder
+   * @param file_name file name
+   * @param offset  number of bytes to offset from beginning of file
```

Comment:
Should note that this parameter is modified

Reply:
@jacobkeeler 
Fixed in:https://github.com/smartdevicelink/sdl_atf/pull/180/commits/2913e68ca6f9435489a6a1e219736c3e85853e8f